metadata-ingestion/src/datahub/ingestion/source/hex/mapper.py

```diff
@@ -0,0 +1,330 @@
+        self._base_url = base_url.strip("/").replace("/api/v1", "")
+        self._patch_metadata = patch_metadata
+        self._collections_as_tags = collections_as_tags
+        self._status_as_tag = status_as_tag
+        self._categories_as_tags = categories_as_tags
+        self._set_ownership_from_email = set_ownership_from_email
+
+    def map_workspace(self) -> Iterable[MetadataWorkUnit]:
+        container_properties = ContainerPropertiesClass(
+            name=self._workspace_name,
+        )
+        aspects: List[_Aspect] = [container_properties]
+        yield from self._yield_mcps(
+            entity_urn=self._workspace_urn,
+            aspects=aspects,
+        )
+
+    def map_project(self, project: Project) -> Iterable[MetadataWorkUnit]:
+        dashboard_urn = self._get_dashboard_urn(name=project.id)
+
+        dashboard_info = DashboardInfoClass(
+            title=project.title,
+            description=project.description or "",
+            lastModified=self._change_audit_stamps(
+                created_at=project.created_at, last_edited_at=project.last_edited_at
+            ),
+            externalUrl=f"{self._base_url}/{self._workspace_name}/hex/{project.id}",
+            customProperties={k: str(v) for k, v in project._raw_data.items()}
+            if project._raw_data
+            else None,
+        )
+
+        subtypes = SubTypesClass(
+            typeNames=[BIAssetSubTypes.HEX_PROJECT],
+        )
+
+        platform_instance = self._platform_instance_aspect()
+
+        container = ContainerClass(
+            container=self._workspace_urn.urn(),
+        )
+
+        aspects = [dashboard_info, subtypes, platform_instance, container]
+
+        tags = self._global_tags(
+            status=project.status,
+            categories=project.categories,
+            collections=project.collections,
+        )
+        if tags:
+            aspects.append(tags)
+
+        ownership = self._ownership(creator=project.creator, owner=project.owner)
+        if ownership:
+            aspects.append(ownership)
+
+        dashboard_usage_statistics = self._dashboard_usage_statistics(
+            analytics=project.analytics
+        )
+        if dashboard_usage_statistics:
+            aspects.append(dashboard_usage_statistics)
+
+        yield from self._yield_mcps(entity_urn=dashboard_urn, aspects=aspects)
+
+    def map_component(self, component: Component) -> Iterable[MetadataWorkUnit]:
+        dashboard_urn = self._get_dashboard_urn(name=component.id)
+
+        dashboard_info = DashboardInfoClass(
+            title=component.title,
+            description=component.description or "",
+            lastModified=self._change_audit_stamps(
+                created_at=component.created_at, last_edited_at=component.last_edited_at
+            ),
+            externalUrl=f"{self._base_url}/{self._workspace_name}/hex/{component.id}",
+            customProperties={k: str(v) for k, v in component._raw_data.items()}
```

Comment:
imo if a field is already mapped to a proper datahub attribute, then it feels redundant to also include it in customProperties

Reply:
I understand and even share the concern about redundancy. Initially, I thought including HTTP response JSON as custom properties might be useful for troubleshooting. However, it's not ideal to clutter the UI with debugging details.

With commit b22b610c36cebfd930ad6f3f4e440aef164e057d, I've removed the raw data. Now, custom properties only display the ID, which isn't visible anywhere else aside from the URN.